portfolio/src/main/java/com/google/sps/servlets/Comments.java

```diff
@@ -0,0 +1,200 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Feedback;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet that returns some example content.
+ */
+@WebServlet("/comments")
+public class Comments extends HttpServlet {
+  private class Comment{
+    public String name;
+    public String email;
+    public String comment;
+
+    public Comment(String name, String email, String comment){
+      this.name = name;
+      this.email = email;
+      this.comment = comment;
+    }
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest request, HttpServletResponse response){
+    // Prepare the Query to store the entity you want to load
+    Query query = new Query("Data");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+	PreparedQuery results = datastore.prepare(query);
+
+    // Delete all the comments in the datastore admin page
+      for(Entity entity : results.asIterable()){
+        datastore.delete(entity.getKey());
+      }
+    }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Prepare the Query to store the entity you want to load
+    Query query = new Query("Data").addSort("Timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+	PreparedQuery results = datastore.prepare(query);
+
+    // Get the input from the form
+    int userChoice = getNumberOfComments(request);
+    if (userChoice == -1) {
+      response.setContentType("text/html");
```

Comment:
Set a failure status in the response: https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#setStatus(int,%20java.lang.String)

Probably a BAD_REQUEST: https://docs.oracle.com/javaee/6/api/javax/servlet/http/HttpServletResponse.html#SC_BAD_REQUEST